CMakeLists.txt

```diff
@@ -27,27 +30,118 @@ if (Git_FOUND)
   endif()
 endif()
 
+macro(set_cxx_flag FLAG)
+	if (${ARGC} GREATER 1)
+		set(CMAKE_CXX_FLAGS_${ARGV1} "${CMAKE_CXX_FLAGS_${ARGV1}} ${FLAG}")
+	else()
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${FLAG}")
+	endif()
+endmacro()
+
+macro(set_linker_flag FLAG)
+	if (${ARGC} GREATER 1)
+		set(CMAKE_EXE_LINKER_FLAGS_${ARGV1} "${CMAKE_EXE_LINKER_FLAGS_${ARGV1}} ${FLAG}")
+		set(CMAKE_SHARED_LINKER_FLAGS_${ARGV1} "${CMAKE_SHARED_LINKER_FLAGS_${ARGV1}} ${FLAG}")
+		set(CMAKE_MODULE_LINKER_FLAGS_${ARGV1} "${CMAKE_MODULE_LINKER_FLAGS_${ARGV1}} ${FLAG}")
+	else()
+		set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${FLAG}")
+		set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${FLAG}")
+		set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} ${FLAG}")
+	endif()
+endmacro()
+
+# This option decides if crunch is dynamically linked against libcrn.so
+# statically linked against libcrn.o, enabling it always build libcrn.so.
+# This option is a builtin CMake one, the name means “build executables
+# against shader libraries”, not “build the shared libraries”.
 option(BUILD_SHARED_LIBS "Link executables against shared library" OFF)
+# Always build libcrn.so even if crunch is linked to libcrn statically.
 option(BUILD_SHARED_LIBCRN "Build shared libcrn" OFF)
+# Always build libcrn.a even if crunch is linked to libcrn dynamically.
 option(BUILD_STATIC_LIBCRN "Build static libcrn" OFF)
+# Build the crunch tool, implies the build of libcrn.o or libcrn.so.
 option(BUILD_CRUNCH "Build crunch" ON)
+# Build the provided examples, they only build on Windows for now.
 option(BUILD_EXAMPLES "Build examples" OFF)
-if (NOT MSVC)
-  option(OPTIMIZE_RELEASE "Optimize release build" ON)
-endif()
+# Enable extra optimization flags, like using -O3 even in RelWithDebInfo build.
+option(USE_EXTRA_OPTIMIZATION "Enable extra optimization" ON)
+# Enable link time optimization, slows down the build but produce faster and smaller binaries.
+option(USE_LTO "Enable link-time optimization" ON)
+# Fast math can be used if generated files are written to a cache, but it is not a good idea
+# to use fast math if the generated files are meant to be distributed and reproducible.
+# See https://github.com/DaemonEngine/crunch/issues/29
+option(USE_FAST_MATH "Enable fast math (files generated with the executable may not be reproducible)" ON)
```

Comment:
You can say that they are less likely to be reproducible. Saying that they are reproducible with this off is too strong of a claim.

Reply:
Reworded.